internal/migrations/s3_layout.go

```diff
@@ -14,15 +14,15 @@ import (
 )
 
 func init() {
-	register(&S3Layout{})
+	register(S3Layout{})
 }
 
 // S3Layout migrates a repository on an S3 backend from the "s3legacy" to the
 // "default" layout.
 type S3Layout struct{}
 
 // Check tests whether the migration can be applied.
-func (m *S3Layout) Check(ctx context.Context, repo restic.Repository) (bool, error) {
+func (m S3Layout) Check(ctx context.Context, repo restic.Repository) (bool, error) {
```

Comment:
I'm not sure why this was changed? Having only pointer methods produces smaller code, because for value methods, the compiler generates stubs for the pointer versions too.

Reply:
Ah, hm. I thought it was unnecessary to have a pointer. I'll revert that. To be honest I'm itching to refactor the migration interface, but I don't think it's worth it...